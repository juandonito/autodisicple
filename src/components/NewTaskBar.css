.NewTaskBar {
    width: 90%;
    margin: auto;

    padding: 1rem 0;

    text-align: center;
}

.NewTaskBar input {
    width: 70%;
    margin: auto;

    font-size: 1.5rem;

    height: 1.5rem;
    border: none;
    border-bottom: 1px solid black;

    outline: none;
}