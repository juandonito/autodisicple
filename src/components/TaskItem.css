.TaskItem {
    font-size: 1.25rem;
    line-height: 2rem;
}

.TaskItem.done {
    text-decoration: line-through 
}