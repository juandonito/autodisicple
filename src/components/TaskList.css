.TaskList {
    width: 90%;
    margin: auto;
}