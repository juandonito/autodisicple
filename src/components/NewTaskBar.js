import './NewTaskBar.css'

import React from 'react'

import { connect } from 'react-redux'
import { doAddTask } from '../actions/tasks'

export class NewTaskBar extends React.Component{

    state = {
        input : ''
    }

    handleInputChange = (e) => {
        this.setState({
            input: e.target.value
        })
    }

    handleInputSubmit = (e) => {
        e.preventDefault();
        this.props.onTaskAdd(this.state.input);
        this.setState({
            input: ''
        })
    }

    render() {

        const { input } = this.state;

        return (
            <div className='NewTaskBar'>
                <form onSubmit={this.handleInputSubmit}>
                    <input 
                        type = 'text' 
                        placeholder = 'New task' 
                        value = {input}
                        onChange = {this.handleInputChange}
                        spellCheck='false'
                    />
                </form>
            </div>
        )

    }

}

const mapDispatchToProps = dispatch => ({
    onTaskAdd: title => dispatch(doAddTask(title))
});

export default connect(null, mapDispatchToProps)(NewTaskBar);