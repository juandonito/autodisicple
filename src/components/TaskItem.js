import './TaskItem.css'

import React from 'react'
import { connect } from 'react-redux'
import { doToggleTask } from '../actions/tasks'

export const TaskItem = ({ task, onToggleTask }) => {
    return (
        <div className='TaskItem' onClick={() => onToggleTask(task.id)}>{task.title}</div>
    )
}

const mapDispatchToProps = dispatch => {
    return {
        onToggleTask: id => dispatch(doToggleTask(id))
    }
}

export default connect(null, mapDispatchToProps)(TaskItem);