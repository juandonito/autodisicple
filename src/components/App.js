import './App.css'

import React from 'react'

import NewTaskBar from './NewTaskBar'
import TaskList from './TaskList'

function App () {
    return (
        <div className='App'>
            <NewTaskBar />
            <TaskList />
        </div>
    )
}

export default App;