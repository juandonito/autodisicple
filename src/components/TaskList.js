import './TaskList.css'

import React from 'react'
import TaskItem from './TaskItem'

import { getTodayTask } from '../selectors/task'

import { connect } from 'react-redux'

export const TaskList = ({ tasks }) => {

    return (
        <div className='TaskList'>
            {
                tasks.map(task => (
                    <TaskItem key={task.id} task={task} />
                ))
            }
        </div>
    )
}

const mapStateToProps = (state) => ({
    tasks:  getTodayTask(state)
})

export default connect(mapStateToProps)(TaskList);