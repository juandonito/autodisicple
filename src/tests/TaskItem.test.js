import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { TaskItem } from '../components/TaskItem'

describe('TaskItem', () => {

    const TASK = {
        title: 'this is a task test',
        id: 1
    }

    it('renders whitout crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TaskItem task={TASK}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    test('has valid snapshot', () => {
        const component = renderer.create(
            <TaskItem task={TASK} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

})