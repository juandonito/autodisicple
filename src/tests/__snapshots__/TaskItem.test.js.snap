// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem has valid snapshot 1`] = `
<div
  className="TaskItem"
  onClick={[Function]}
>
  this is a task test
</div>
`;
